cmd/tdex/feeder.go

```diff
@@ -0,0 +1,305 @@
+				Usage: "ticker of the market, e.g. XBT/USDT, XBT/EUR etc",
+			},
+		},
+	}
+	removePriceFeed = &cli.Command{
+		Name:   "remove",
+		Usage:  "remove a price feed",
+		Action: removePriceFeedAction,
+		Flags: []cli.Flag{
+			&cli.StringFlag{
+				Name:  "id",
+				Usage: "id of the price feed to start",
+			},
+		},
+	}
+	getPriceFeed = &cli.Command{
+		Name:   "get",
+		Usage:  "get a price feed",
+		Action: getPriceFeedAction,
+		Flags: []cli.Flag{
+			&cli.StringFlag{
+				Name:  "base_asset",
+				Usage: "base asset of the market for which the price feed is created",
+			},
+			&cli.StringFlag{
+				Name:  "quote_asset",
+				Usage: "quote asset of the market for which the price feed is created",
+			},
+		},
+	}
+	listPriceFeeds = &cli.Command{
+		Name:   "list",
+		Usage:  "list price feeds",
+		Action: listPriceFeedsAction,
+	}
+	listSources = &cli.Command{
+		Name:   "sources",
+		Usage:  "list price feed sources",
+		Action: listSourcesAction,
+	}
+)
+
+func addPriceFeedAction(ctx *cli.Context) error {
+	client, cleanup, err := getFeederClient(ctx)
+	if err != nil {
+		return err
+	}
+	defer cleanup()
+
+	baseAsset := ctx.String("base_asset")
+	quoteAsset := ctx.String("quote_asset")
+	source := ctx.String("source")
+	ticker := ctx.String("ticker")
+
+	if baseAsset == "" || quoteAsset == "" || source == "" || ticker == "" {
+		return cli.Exit("base_asset, quote_asset, source and ticker are required", 1)
+	}
+
+	if _, err := client.AddPriceFeed(ctx.Context, &daemonv2.AddPriceFeedRequest{
+		Market: &tdexv2.Market{
+			BaseAsset:  baseAsset,
+			QuoteAsset: quoteAsset,
+		},
+		Source: source,
+		Ticker: ticker,
+	}); err != nil {
+		return cli.Exit(err, 1)
+	}
+
+	fmt.Println("Price feed added")
+	return nil
+}
+
+func startPriceFeedAction(ctx *cli.Context) error {
+	client, cleanup, err := getFeederClient(ctx)
+	if err != nil {
+		return err
+	}
+	defer cleanup()
+
+	id := ctx.String("id")
+
+	if id == "" {
+		return cli.Exit("id is required", 1)
+	}
+
+	if _, err := client.StartPriceFeed(ctx.Context, &daemonv2.StartPriceFeedRequest{
+		Id: id,
+	}); err != nil {
+		return cli.Exit(err, 1)
+	}
+
+	fmt.Println("Price feed started")
+	return nil
+}
+
+func stopPriceFeedAction(ctx *cli.Context) error {
+	client, cleanup, err := getFeederClient(ctx)
+	if err != nil {
+		return err
+	}
+	defer cleanup()
+
+	id := ctx.String("id")
+	if id == "" {
+		return cli.Exit("id is required", 1)
+	}
+
+	if _, err := client.StopPriceFeed(ctx.Context, &daemonv2.StopPriceFeedRequest{
+		Id: id,
+	}); err != nil {
+		return cli.Exit(err, 1)
+	}
+
+	fmt.Println("Price feed stopped")
+	return nil
+}
+
+func updatePriceFeedAction(ctx *cli.Context) error {
+	client, cleanup, err := getFeederClient(ctx)
+	if err != nil {
+		return err
+	}
+	defer cleanup()
+
+	id := ctx.String("id")
+	source := ctx.String("source")
+	ticker := ctx.String("ticker")
+	if id == "" || source == "" || ticker == "" {
+		return cli.Exit("id, source and ticker are required", 1)
+	}
```

Comment:
in this specific command only `id` must be required, `source` and `ticker` are optional instead, since the operator might want to change only the ticker because he used the wrong one for example.

Also, it should be responsibility of the daemon to return an error in case the operator passes empty source and ticker in the message request. At least one of the two must not be empty. This check should be done at infra level I think.

Reply:
Done